[mongj] iP

mongj opened the pull request:
# iP  
> This is a quote

This bot frees your mind of having to remember things you need to do. It's,

* text-based  
* easy to learn  
* ~FAST~ SUPER FAST to use  

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).  
2. double-click it.  
3. add your tasks.  
4. let it manage your tasks for you 😉  

**And it is FREE!**

Features:

* [x] Managing tasks  
* [ ] Managing deadlines (coming soon)  
* [ ] Reminders (coming soon)  

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

mongj committed:
Set up ChadGPT

mongj committed:
Add printBotMessage method

mongj committed:
Add Formatter class

mongj committed:
Add ability to echo commands

mongj committed:
Add ability to store and list tasks

mongj committed:
Add ability to mark tasks

mongj committed:
Refactor command handling

mongj committed:
Add new task types (todo, deadline, event)

mongj committed:
Update ui-test target from Duke to ChadGPT

mongj committed:
Refactor project into packages

mongj committed:
Implement basic exception handling

mongj committed:
Implement task deletion

mongj committed:
Use enums for commands

mongj committed:
Add local file storage for tasks

mongj committed:
Fix code formatting

mongj committed:
Merge branch 'branch-Level-7'

mongj committed:
Use LocalDate for datetime

mongj committed:
Merge branch 'branch-Level-8'

mongj committed:
Add new TaskList abstraction

mongj committed:
Add new Ui abstraction

mongj committed:
Add basic parser abstraction

mongj committed:
Merge remote-tracking branch 'origin/add-gradle-support'

mongj committed:
Add basic Gradle support

mongj committed:
Add basic unit tests for Parser class

mongj committed:
Add basic unit tests for TaskList class

mongj committed:
build.gradle: Specify jar filename

mongj committed:
Add JavaDoc comments for some methods

mongj committed:
Fix formatting according to Coding standard

mongj committed:
Add basic search function

mongj committed:
Merge branch 'A-CodingStandard'

mongj committed:
Merge branch 'Level-9'

mongj committed:
Remove debug print statements

jonushzw commented on src/main/java/bot/Bot.java:
```diff
@@ -0,0 +1,218 @@
+        }
+    }
+
+    private void handleList() {
```
Could rename to displayList or displayTaskList to show that it displays, instead of just dealing with a generic list

jonushzw commented on src/main/java/bot/Bot.java:
```diff
@@ -0,0 +1,218 @@
+        ui.printMessage("OK, I've marked this task as not done yet:\n" + unmarkedTask);
+    }
+
+    private int getTaskIndex(String input) {
```
Could be renamed to parseTaskIndexFromInput to indicate that the method parses the task index from user input.

jonushzw commented on src/main/java/bot/Bot.java:
```diff
@@ -0,0 +1,218 @@
+            return;
+        }
+
+        Command cmd = parsedInput.getCmd();
```
spelling the full word out could be better. Just a suggestion

jonushzw commented on src/main/java/bot/Parser.java:
```diff
@@ -0,0 +1,90 @@
+public class Parser {
+    // TODO: Move this into a Command class
+    public static class ParsedInput {
+        private final Command cmd;
```
same here could use the full name

jonushzw commented on src/main/java/bot/Ui.java:
```diff
@@ -0,0 +1,28 @@
+     * Prints a welcome message.
+     */
+    public void start() {
+        printMessage("Hello! I'm ChadGPT. What can I do for you?");
```
Hahaha. You're so funny! XD

jonushzw commented on src/main/java/bot/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+     * If the directory or file(s) cannot be initialised, the program exits.
+     */
+    private void init() {
+        File d = new File(DIR_PATH);
```
could rename d to `directory` to be more specific and improve readability king!

jonushzw commented on src/main/java/bot/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+
+        // TODO: Abstract storage of each type into its own class
+        // Initialize task storage
+        File f = new File(TASK_FILE_PATH);
```
same here could rename it to `taskFile` for better readibility

jonushzw commented on src/main/java/bot/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+     * @throws IOException If file cannot be saved.
+     */
+    public void saveTaskList(TaskList tasks) throws IOException {
+        FileWriter fw = new FileWriter(TASK_FILE_PATH);
```
rename to `fileWriter` for better clarity

jonushzw reviewed (commented):
Upz Upz Upz bro! Keep up the good work king!

gabriellegtw commented on src/main/java/bot/tasks/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+ * @author mongj
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks = new ArrayList<>();
```
I like how you followed naming conventions for java collections :)

gabriellegtw commented on src/main/java/bot/tasks/Task.java:
```diff
@@ -0,0 +1,47 @@
+        this.isDone = false;
+    }
+
+    public Task(String description, boolean isDone) {
```
Great that you followed naming conventions for booleans

gabriellegtw commented on src/main/java/bot/tasks/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+
+    public Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by;
```
Perhaps a more descriptive variable name could be used here?

gabriellegtw commented on src/main/java/bot/tasks/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+        return tasks.get(index);
+    }
+
+    /**
```
I like the descriptive java docs throughout the code

gabriellegtw reviewed (commented):
LGTM. Coding standard wise, not much to fix or add!

mongj committed:
Add checkstyle

mongj committed:
Implement GUI for chatbot

mongj committed:
Enable assertions in build.gradle

Java disables assertions by default. However, we want to use assertions
in our code to make sure invariants are maintained at runtime.

Let's update the gradle build config to enable assertions.

mongj committed:
Use Java assertions when parsing stored data

File data can be corrupted and there is a need to validate
the data to prevent unforeseen runtime errors.

Minimally, let's add assertions for the parseData method
in the Storage class to check that the data read has the
correct number of arguments.

mongj committed:
Fix import order for all .java files

Checkstyle is currently complaining about the order
of our import statements, namely:
* java.* imports should come before the rest, and in a separate block
* all imports should be sorted in lexicographical order

Let's fix the import ordering issue for all existing .java files

mongj committed:
Remove unused Formatter class

Formatter class was used when we had text-based UI.
It is no longer used as we have transitioned over to the new GUI.

Let's remove the Formatter class to make the code cleaner.

mongj committed:
Remove unused DateTimeParseException

mongj committed:
Use all uppercase letter for constant INTRO

mongj committed:
Fix spacing issues

mongj committed:
Add header for parseData

mongj committed:
Add TaskSymbol enum abstraction

Symbols for tasks like "T", "D", "E" are reused
in multiple places in the code.

To improve maintainability, let's abstract these
symbols out into a common reusable enum.

mongj committed:
Merge pull request #1 from mongj/A-Assertions

Use Java Assertions where applicable

mongj committed:
Merge A-Assertions into A-CodeQuality

mongj committed:
Merge pull request #2 from mongj/A-CodeQuality

Refactor code to improve code quality

mongj committed:
Refactor actions and add undo functionality

mongj committed:
Add GitHub Actions yml config

mongj committed:
Remove IO redirection checks from gradle.yml

mongj committed:
Fix minor JavaDoc issues

mongj committed:
Fix checkstyle issues in main

mongj committed:
Fix checkstyle issues in test

mongj committed:
Suppress checkstyle warnings for switch case indentation

mongj committed:
Update user and bot profile pictures

mongj committed:
Add stage title and icon

mongj committed:
Add basic user guide

mongj committed:
Rename app.png to Ui.png to follow submission requirement

mongj committed:
Remove unnecessary import group separation

mongj committed:
Refactor bot name

mongj committed:
Add exception handling for datetime parsing

mongj committed:
Refactor exception messages

mongj committed:
Add check for empty message

mongj committed:
Refactor bot exception message

mongj committed:
Fix incorrect exception in test

mongj committed:
Use local parseDate function

mongj committed:
Fix typo in doc

mongj committed:
Add help command

mongj committed:
Add parser for todo task

mongj committed:
Fix incorrect exit status code

mongj committed:
Fix minor styling issues

mongj committed:
Improve doc details

mongj committed:
Improve doc details
